File: Selenium/Project/SearchAndApplyForJob.py
import unittest
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class SearchAndApplyForJob(unittest.TestCase):
    def setUp(self):
        self.driver = webdriver.Firefox(executable_path="C:/Users/001ZXG744/IdeaProjects/geckodriver.exe")
        self.base_url = "https://alchemy.hguy.co/jobs/"

    def test_search_and_apply_for_job(self):
        # Navigate to the Jobs page
        self.driver.get(self.base_url)
        jobs_menu_item = self.driver.find_element(By.LINK_TEXT, "Jobs")
        jobs_menu_item.click()

        # Search for a particular job
        search_input = self.driver.find_element(By.ID, "search_keywords")
        search_input.send_keys("Software Engineer")

        # Click the Search button
        search_button = self.driver.find_element(By.XPATH, "//input[@type='submit']")
        search_button.click()

        # Wait for job listings to show
        wait = WebDriverWait(self.driver, 10)
        wait.until(EC.visibility_of_element_located((By.CLASS_NAME, "job_listing")))

        # Click and open any one of the jobs listed
        job_listing = self.driver.find_element(By.CLASS_NAME, "job_listing")
        job_listing.click()

        # Click the apply button
        apply_button = self.driver.find_element(By.CLASS_NAME, "application_button")
        apply_button.click()

        # Print the email to the console
        email_element = self.driver.find_element(By.XPATH, "//div[@class='job_application_email']")
        email = email_element.text
        print("Job Application Email: " + email)

    def tearDown(self):
        self.driver.quit()

if __name__ == "__main__":
    unittest.main()

File: Selenium/Project/SearchAndApplyForJob.java
package Project;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class SearchAndApplyForJob {
        private WebDriver driver;
        private String baseUrl = "https://alchemy.hguy.co/jobs/";

        @BeforeClass
        public void setUp() {
            System.setProperty("webdriver.gecko.driver",  "C://Users//001ZXG744//IdeaProjects//geckodriver.exe");
            driver = new FirefoxDriver();
            driver.get(baseUrl);
        }

        @Test
        public void testSearchAndApplyForJob() {
            // Navigate to the Jobs page
            WebElement jobsMenuItem = driver.findElement(By.linkText("Jobs"));
            jobsMenuItem.click();

            // Search for a particular job
            WebElement searchInput = driver.findElement(By.id("search_keywords"));
            searchInput.sendKeys("Software Engineer");

            // Click the Search button
            WebElement searchButton = driver.findElement(By.xpath("//input[@type='submit']"));
            searchButton.click();

            // Wait for job listings to show
            WebDriverWait wait = new WebDriverWait(driver, 10);
            wait.until(ExpectedConditions.visibilityOfElementLocated(By.className("job_listing")));

            // Click and open any one of the jobs listed
            WebElement jobListing = driver.findElement(By.className("job_listing"));
            jobListing.click();

            // Click the apply button
            WebElement applyButton = driver.findElement(By.className("application_button"));
            applyButton.click();

            // Print the email to the console
            WebElement emailElement = driver.findElement(By.xpath("//div[@class='job_application_email']"));
            String email = emailElement.getText();
            System.out.println("Job Application Email: " + email);
        }

        @AfterClass
        public void tearDown() {
            driver.quit();
        }
    }

